phoenix-core/src/main/java/org/apache/phoenix/iterate/TableSnapshotResultIterator.java

```diff
@@ -88,19 +90,34 @@ public TableSnapshotResultIterator(Configuration configuration, Scan scan, ScanM
   }
 
   private void init() throws IOException {
-    RestoreSnapshotHelper.RestoreMetaChanges meta =
-        RestoreSnapshotHelper.copySnapshotForScanner(this.configuration, this.fs,
-            this.rootDir, this.restoreDir, this.snapshotName);
-    List<RegionInfo> restoredRegions = meta.getRegionsToAdd();
-    this.htd = meta.getTableDescriptor();
-    this.regions = new ArrayList<RegionInfo>(restoredRegions.size());
-
-    for (RegionInfo restoredRegion : restoredRegions) {
-      if (isValidRegion(restoredRegion)) {
-        this.regions.add(restoredRegion);
+    if (PhoenixConfigurationUtil.getMRSnapshotManagedInternally(configuration)) {
```

Comment:
Not sure I get this, We are again restoring per iterator which is per scan? Isn't this what caused the issue in the first place. I see that it is guarded by a configuration option but ideally we want to avoid this in all cases right? or to rephrase my question, who wants to set this to true?

Reply:
@bharathv doing the restore per MR job can be handled at one place but I was not able to get any spot to delete/cleanup the restore right after the job is done on phoenix side. On the caller side, I am able to do the cleanup, so in order to handle it in a clean way: I added this configuration to explicitly say: whether or not restore needs to be managed internally or externally. If internally I am keeping the original flow (with issue) as it is. Externally: restore + delete happens on the MR job caller side.